### PR TITLE
Improve coverage on edge cases + misc. testing

### PR DESCRIPTION
**Pull request by nbgl**
- Improve test coverage on edge cases.
- Minor changes to code to make those tests pass, including slight changes to exception throwing.
- revert `test_similarity.py` to previous, so we still test the old Dice coefficient API
- rename tests for my Dice and Hamming similarities to `test_similarity_hamming.py` and `test_similarity_dice.py` respectively
- switch some tests from unittest to pytest

### CODE REVIEW

**Comment by nbgl** [comment]
Tests are failing because of a bug that exists in `develop`. The fix is in the branch `fix-zero-popcount`. I am waiting for #126 to be reviewed and merged before I can make that pull request.

File: tests/test_similarity.py [hunk 1]
```diff
@@ -1,15 +1,14 @@
+
```
Comment (nbgl) [inline tests/test_similarity.py]: This whole file was reverted to the pre-multiparty version. This is so the PR I make later is strictly additive and we can check that it doesn’t break anything.

The only change I made from the pre-multiparty version is fixing the interaction with Clkhash so it uses the newer API.

File: tests/test_similarity_dice.py [hunk 2]
```diff
@@ -0,0 +1,253 @@
+import unittest
```
Comment (nbgl) [inline tests/test_similarity_dice.py]: This file is mostly the same as test_similarity.py, but it uses the new API instead of the old API.

File: tests/test_blocking.py [hunk 3]
```diff
@@ -3,10 +3,13 @@
 import random
 import unittest
 
+import bitarray
+import pytest
```
Comment (nbgl) [inline tests/test_blocking.py]: In addition to adding tests, I also modified this file to use Pytest instead of Unittest.
